processing/data_collection/gazette/spiders/pr_foz_do_iguacu.py

```diff
@@ -0,0 +1,71 @@
+import datetime as dt
+
+import scrapy
+from scrapy.http import FormRequest
+
+from gazette.items import Gazette
+
+
+class PrFozDoIguacuSpider(scrapy.Spider):
+    MUNICIPALITY_ID = '4108304'
+    name = 'pr_foz_do_iguacu'
+    allowed_domains = ['pmfi.pr.gov.br']
+    start_urls = ['http://www.pmfi.pr.gov.br/utilidades/diario/index.xhtml']
+    AVAILABLE_FROM = dt.date(2015, 1, 1)  # actually September/2007
+
+    def parse(self, response):
+        """
+        @url http://www.pmfi.pr.gov.br/utilidades/diario/index.xhtml
+        @returns requests 1 1
+        """
+        selector = '(//span[@class="ui-paginator-current"])[1]/text()'
+        paginator_text = response.xpath(selector).extract_first()
+        quantity_of_documents = int(paginator_text[1:-1].split()[-1]) + 1
+
+        data = {
+            'formDiarioOficialView': 'formDiarioOficialView',
+            'formDiarioOficialView:dtdiario': 'formDiarioOficialView:dtdiario',
+            'formDiarioOficialView:dtdiario_encodeFeature': 'true',
+            'formDiarioOficialView:dtdiario_first': '0',
+            'formDiarioOficialView:dtdiario_pagination': 'false',
+            'formDiarioOficialView:dtdiario_rows': str(quantity_of_documents),
+        }
+        return FormRequest(response.url, formdata=data, callback=self.parse_items)
+
+    def parse_items(self, response):
+        """
+        @url http://www.pmfi.pr.gov.br/utilidades/diario/index.xhtml
+        @returns items 10 10
+        """
+        base_url = 'http://www.pmfi.pr.gov.br{}'
+        lines = response.xpath('//tr[@role="row"]')
```

Comment:
What about moving the CSS and XPATH queries to constants? In this case it would be:

```python
ROW_XPATH = '//tr[@role="row"]'

...

lines = response.xpath(self.ROW_XPATH)

Reply:
Hey @alfakini, I don't know any scrapy best practices and I didn't find any content related to this matter, so sorry if I am saying some nonsense.
I don't see the benefit from creating constants of selectors, they are part of the logic, used just one time and strongly related to the manipulations around it (different from urls/domains). And in the case of this project, they will be mixed with spider identification, what could be harmful to readability.

But, as I said, I don't know anything about better practices of scrapy spiders and I'm completely open to suggestions =)

Thanks for reviewing it

Reply:
I work with Scrapy on a daily basis, and I don't have the habit to create constants for the selectors, as you said, are used only once. IMO this is more a personal style choice than a best practice.